src/components/UserInfo.js

```diff
@@ -1,39 +1,31 @@
 class UserInfo {
-  constructor({ nameSelector, descriptionSelector }) {
+  constructor({ nameSelector, descriptionSelector, avatarSelector }) {
     this._nameElement = document.querySelector(nameSelector);
     this._descriptionElement = document.querySelector(descriptionSelector);
+    this._avatarElement = document.querySelector(avatarSelector);
   }
 
+  // User Current Info getter
   getUserInfo() {
     return {
       name: this._nameElement.textContent,
       description: this._descriptionElement.textContent,
+      avatar: this._avatarElement.src,
     };
   }
 
-  setUserInfo({ name, description }) {
+  // User's new info setting
+  setUserInfo({ name, description, avatar }) {
     this._nameElement.textContent = name;
     this._descriptionElement.textContent = description;
+    if (avatar) {
+      this._avatarElement.src = avatar;
```

Comment:
Please, use `setNewAvatar` here not to duplicate the code

Reply:
When I change this one it breaks, because in the index.js  I set the user's data using setUserInfo method and take the avatar from the userData passed, checking if the avatar has been provided ( for the case of profile data update without change of avatar ). If I put the method there instead of the line it breaks. The method is for avatar change only.

```js
api
  .loadPageContent()
  .then(([initialCards, userData]) => {
    userInfo.setUserInfo({
      name: userData.name,
      description: userData.about,
      avatar: userData.avatar,
    });
```

Reply:
> The method is for avatar change only.


```js

  // User's new info setting
  setUserInfo({ name, description, avatar }) {
    this._nameElement.textContent = name;
    this._descriptionElement.textContent = description;
    if (avatar) {
      this.setNewAvatar({ avatar });  // this is the way you should reuse `setNewAvatar`
    }
  }
```


`{}` should be used to make an object with the correct data